app/models/session_type.rb

```diff
@@ -1,11 +1,13 @@
 class SessionType < ActiveRecord::Base
   belongs_to :event
   has_many :sessions
+  has_many :proposals
 
   validates_presence_of :name, :event
   validates_uniqueness_of :name, scope: :event
 
   scope :sort_by_name, ->{ order(:name) }
+  scope :public_types, ->{ where(public: true)}
```

Comment:
Unfortunately, this is one of those rare cases. Rails already defines "public" on a model, so I had to go with public_types.